.github/workflows/docs.yml

```diff
@@ -56,10 +58,7 @@ jobs:
           echo "<meta name=\"robots\" content=\"noindex\">" > header.html
 
       - name: Build docs
-        env:
-          # needs to be in sync with [package.metadata.docs.rs]
-          RUSTDOCFLAGS:  -Zunstable-options --cfg=docsrs
-        run: cargo doc --all-features --no-deps -p bevy -Zunstable-options -Zrustdoc-scrape-examples
+        run: cargo docs-rs -p bevy
```

Comment:
Does this end up building all the subcrates as well, and link them together using `extern-html-root-url` or does it just build the `bevy` crate? Otherwise the result will still be different than on docs.rs (e.g. the `source` links won't work). Not that this works right now though...

Reply:
It doesn't link them when running locally. So I'm guessing the source link still won't work.

Reply:
That's unfortunate. By looking at the documentation for `cargo docs-rs` it doesn't seem to support a `--workspace` option either, which I was planning to use in https://github.com/bevyengine/bevy/pull/12965

Reply:
@SkiFire13 what specifically does the `--workspace` option do that is needed for source definitions? I can try opening a PR to `cargo-docs-rs` if I understand your use-case.

Reply:
Source definitions are part of the documentation of the dependencies, which thus need to be built as well or rustdoc will omit the source definition links (because it has nowhere to link to).

Our `docs` workflow does not build dependencies (notice the `--no-deps` flag), and thus source links are unavailable. A simple solution would be removing that option, however you then have to build the documentation for all our 400-500 dependencies (not just the `bevy` crates!). A cheaper alternative is using `--workspace`, which just tells cargo to build the documentation for all the crates in the workspace while still skipping the documentation for other dependencies (for which source links will still not be available, though this is much less of an issue).

docs.rs does not have this problem because it has to build the documentation for all dependencies anyway when those are published, though I think they also need to do something weird with `extern-html-root-url`/`-Zrustdoc-map` to properly link them (which we could also use to link to our dependencies, though of course this doesn't work for the `bevy_*` crates which are not yet released on docs.rs).

Reply:
this isn't a brilliant solution, but you should just be able to run `cargo docs-rs` multiple times for each crate we want to document. (i've verified locally that) sequential operations should update the relevant files.

Reply:
of course it would be nice to upstream the `--workspace` flag to `cargo docs-rs` but this seems to work too.